packages/editor/src/lib/editor/Editor.ts

```diff
@@ -5733,6 +5724,21 @@ export class Editor extends EventEmitter<TLEventMap> {
 				}
 			)
 
+			orderedShapeIds.forEach((originalId, i) => {
+				const originalShape = this.getShape(originalId)
+				if (!originalShape) return
+				const parentId = originalShape.parentId
+				const siblings = this.getSortedChildIdsForParent(parentId)
+				const currentIndex = siblings.indexOf(originalShape.id)
+				const siblingAboveId = siblings[currentIndex + 1]
+				const siblingAbove = siblingAboveId ? this.getShape(siblingAboveId) : null
+
+				const index = siblingAbove
+					? getIndexBetween(originalShape.index, siblingAbove.index)
```

Comment:
Its ok for the second parameter here to be undefined, so you could do 

`index = getIndexBetween(originalShape.index, this.getShape(siblings[currentIndex + 1]))`

Reply:
Updated this. Still had to do some checks so that we don't call `this.getShape` with `undefined`.